### PR TITLE
[MM-36235] - Do not allow to downgrade cloud plans

### PR DESCRIPTION
**Pull request by AGMETEOR**
#### Summary
Currently, if the user is subscribed in the Starter or Professional, when the admin clicks the "Upgrade" button the purchase modal shows all options. They should only be presented with the higher options to upgrade (and the current plan for comparison)

#### Ticket Link
[MM-36235](https://mattermost.atlassian.net/browse/MM-36235?atlOrigin=eyJpIjoiMTA2MDE1MTQ1MTcwNGJjMDgyYzYyMmViMWRjNmE3MjMiLCJwIjoiaiJ9)

#### Related PRs
[webapp](https://github.com/mattermost/mattermost-webapp/pull/8253)

#### Release Note
```release-note
none
```

### CODE REVIEW

**Review by nickmisasi — APPROVED** [review]

File: components/purchase_modal/purchase_modal.tsx [hunk 1]
```diff
@@ -192,7 +192,9 @@ export default class PurchaseModal extends React.PureComponent<Props, State> {
     listPlans = () => {
         const products = this.props.products!;
         const options = Object.keys(products).map((key: string) => {
-            return {key: products[key].name, value: products[key].id};
+            return {key: products[key].name, value: products[key].id, price: products[key].price_per_seat};
+        }).sort((a, b) => a.price - b.price).filter((option) => {
+            return option.price >= this.state.currentProduct!.price_per_seat;
```
Comment (marianunez) [inline components/purchase_modal/purchase_modal.tsx:197]: @AGMETEOR this code has now changed to consider the `flat_fee` subscriptions in this PR: https://github.com/mattermost/mattermost-webapp/pull/8341/files#diff-acb502910cc86e76a73b356eba7ef5024d0db3aaa31441bbe70ff3c83dc164beR193-R206

We can now update this PR and consider not allowing them to downgrade but *ONLY* when you are upgrading (not the initial purchase from trial given that they may be in Enterprise and can select Starter to subscribe)

**Review by marianunez — CHANGES_REQUESTED** [review]

File: components/purchase_modal/purchase_modal.tsx (hunk 1 above)
Comment (AGMETEOR) [reply components/purchase_modal/purchase_modal.tsx]: I have updated the code

**Review by pablovelezvidal — APPROVED** [review]
LGTM!

**Comment by AGMETEOR** [comment]
@stevemudie I think this is a post merge QA review candidate

**Review by stevemudie — APPROVED** [review]
Will test post merge

File: components/purchase_modal/purchase_modal.tsx [hunk 2]
```diff
@@ -184,26 +192,36 @@ export default class PurchaseModal extends React.PureComponent<Props, State> {
...
         this.setState({selectedProduct: selectedPlan});
     }
 
-    listPlans = () => {
+    listPlans = (): JSX.Element => {
         const products = this.props.products!;
-        const flatFeeProducts: any = [];
-        const userBasedProducts: any = [];
+        const flatFeeProducts: ProductOptions = [];
+        const userBasedProducts: ProductOptions = [];
```
Comment (marianunez) [inline components/purchase_modal/purchase_modal.tsx:204]: Nice 👍

**Review by marianunez — APPROVED** [review]
Great work @AGMETEOR!

**Pull request merged**